pkg/importer/http-datasource.go

```diff
@@ -324,18 +333,25 @@ func getContentLength(client *http.Client, ep *url.URL, accessKey, secKey string
 		klog.V(3).Infof("GO CLIENT: key: %s, value: %s\n", k, v)
 	}
 
+	total := parseHTTPHeader(resp)
+
+	err = resp.Body.Close()
+	if err != nil {
+		return uint64(0), errors.Wrap(err, "could not close head read")
+	}
+	return total, nil
+}
+
+func parseHTTPHeader(resp *http.Response) (uint64) {
+	var err error
 	total := uint64(0)
 	if val, ok := resp.Header["Content-Length"]; ok {
 		total, err = strconv.ParseUint(val[0], 10, 64)
 		if err != nil {
-			return uint64(0), errors.Wrap(err, "could not convert content length")
+			klog.Errorf("could not convert content length, got %d", err)
```

Comment:
%d is not an err compatible replacement, use %v for complex types. %v prints values in a 'default' format. You can also use %s and do err.Error()